monkey/tests/unit_tests/infection_monkey/network/relay/test_utils.py

```diff
@@ -57,7 +59,9 @@ def test_find_available_island_apis__multiple_successes(island_api_client_factor
     available_servers = [SERVER_2, SERVER_3]
     with requests_mock.Mocker() as mock:
         mock.get(f"https://{SERVER_1}/api?action=is-up", exc=IslandAPIConnectionError)
+        mock.post(f"https://{SERVER_1}/api/agent-otp-login", json={"token": "fake-token"})
```

Comment:
If "?action-is-up" raised an `IslandAPIConnectionError`, why is the `agent-otp-login` call being made?